### ISSUE TITLE
Variables are incorrectly marked as shadowed

### ISSUE THREAD

**Issue by domoritz** [eslint/eslint#13649]
**Tell us about your environment**

* **ESLint Version:** v7.8.1

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**

`@typescript-eslint/parser`

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  parser: "@typescript-eslint/parser",
  plugins: ["@typescript-eslint", "prettier"],
  extends: [
    "eslint:recommended",
    "plugin:@typescript-eslint/recommended",
    "prettier",
    "prettier/@typescript-eslint",
    "prettier/standard"
  ],
  env: {
    browser: true,
    node: true
  },
  parserOptions: {
    ecmaVersion: 2018,
    sourceType: "module"
  },
  rules: {
    "@typescript-eslint/explicit-function-return-type": "off",
    "@typescript-eslint/no-explicit-any": "off",
    "@typescript-eslint/prefer-for-of": "error",
    "@typescript-eslint/no-require-imports": "error",
    "@typescript-eslint/no-parameter-properties": "off",
    "no-unused-vars": "off",
    "@typescript-eslint/no-unused-vars": [
      "error",
      {
        vars: "all",
        args: "none",
        ignoreRestSiblings: true
      }
    ],
    "@typescript-eslint/no-object-literal-type-assertion": "off",
    "@typescript-eslint/no-namespace": "off",
    "@typescript-eslint/no-non-null-assertion": "off",
    "max-len": ["error", { code: 120, tabWidth: 4 }],
    "linebreak-style": ["error", "unix"],
    "no-alert": "error",
    "prefer-const": "error",
    "no-return-assign": "error",
    "no-useless-call": "error",
    "no-shadow": "error",
    "no-useless-concat": "error",
    "no-undef": "off",
    "no-prototype-builtins": "off",
    "prettier/prettier": "error"
  }
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```ts
export function foo(bar: number, baz: (bar: number) => void) {}
```

```bash

```

**What did you expect to happen?**

no error

**What actually happened? Please include the actual, raw output from ESLint.**

```
'bar' is already declared in the upper scope.eslintno-shadow
```

**Are you willing to submit a pull request to fix this bug?**

no

**Comment by mdjermanovic** [comment]
Hi @domoritz, thanks for the issue!

Can you try with this rule:

https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-shadow.md 

```json
{
  "no-shadow": "off",
  "@typescript-eslint/no-shadow": ["error"]
}
```

**Comment by domoritz** [comment]
Same issue

<img width="815" alt="Screen Shot 2020-09-01 at 15 29 02" src="https://user-images.githubusercontent.com/589034/91912384-df474d00-ec67-11ea-9fdb-d2b660770e20.png">

**Comment by mdjermanovic** [comment]
Looks like a false positive.

Can you please open an issue about the `@typescript-eslint/no-shadow` rule in the typescript-eslint repository:

https://github.com/typescript-eslint/typescript-eslint

**Comment by domoritz** [comment]
In https://github.com/typescript-eslint/typescript-eslint/issues/2469

**Comment by domoritz** [comment]
Closing since the issue is tracked in https://github.com/typescript-eslint/typescript-eslint/issues/2447 already.
